easybuild/framework/easyblock.py

```diff
@@ -4849,3 +4944,36 @@ def make_checksum_lines(checksums, indent_level):
             ectxt = regex.sub('\n'.join(exts_list_lines), ectxt)
 
         write_file(ec['spec'], ectxt)
+
+
+def update_exts_list(ecs):
+    """
+    Write a new EasyConfig recipe with all extensions in exts_list updated to the latest version.
+
+    :param ecs: list of EasyConfig instances to complete dependencies for
+    """
+
+    for ec in ecs:
+
+        # get the EasyBlock instance
+        print_msg("Getting easyblock instance...", log=_log)
+        app: EasyBlock = get_easyblock_instance(ec)
+
+        # initialize extension instances
+        print_msg("Initializing extension list...", log=_log)
+        app.init_ext_instances()
```

Comment:
```
source /cvmfs/software.eessi.io/versions/2023.06/init/bash (to provide an appstack)
source python-env/bin/activate (to provide requests in a venv)
export PYTHONPATH=/home/dgonzalez/hpc-playground/git/easybuild-framework_2024:$PYTHONPATH
export PYTHONPATH=$PYTHONPATH:/home/dgonzalez/hpc-playground/easybuild-easyconfigs:/home/dgonzalez/hpc-playground/easybuild-easyblocks
cd /home/dgonzalez/hpc-playground/tests
eb R-4.4.1-gfbf-2023b.eb --update-exts-list --force
```
```
ERROR: Improper default extension class specification, should be string: None (<class 'NoneType'>)
```

Reply:
This can be solved by adding something like (pseudocode):
```
If cfg.name=="R" 
define ext_class_specification as RPackage
```

Reply:
Solved in latest commit. Pls review the whole code again